x-pack/plugins/observability_solution/inventory/public/hooks/use_discover_redirect.ts

```diff
@@ -5,52 +5,36 @@
  * 2.0.
  */
 import {
-  ENTITY_TYPE,
   ENTITY_DEFINITION_ID,
   ENTITY_DISPLAY_NAME,
   ENTITY_LAST_SEEN,
+  ENTITY_TYPE,
 } from '@kbn/observability-shared-plugin/common';
 import { useCallback } from 'react';
-import { type PhrasesFilter, buildPhrasesFilter } from '@kbn/es-query';
-import type { DataViewField } from '@kbn/data-views-plugin/public';
 import type { Entity, EntityColumnIds } from '../../common/entities';
 import { unflattenEntity } from '../../common/utils/unflatten_entity';
 import { useKibana } from './use_kibana';
-import { useInventoryParams } from './use_inventory_params';
-import { useInventorySearchBarContext } from '../context/inventory_search_bar_context_provider';
+import { useUnifiedSearchContext } from './use_unified_search_context';
 
 const ACTIVE_COLUMNS: EntityColumnIds[] = [ENTITY_DISPLAY_NAME, ENTITY_TYPE, ENTITY_LAST_SEEN];
 
 export const useDiscoverRedirect = () => {
   const {
     services: { share, application, entityManager },
   } = useKibana();
-  const {
-    query: { kuery, entityTypes },
-  } = useInventoryParams('/*');
 
-  const { dataView } = useInventorySearchBarContext();
+  const { dataView, searchState } = useUnifiedSearchContext();
```

Comment:
Nit: We can define and use the values we need from the `searchState`: 
```suggestion
  const { dataView, searchState } = useUnifiedSearchContext();
  const { query, filters, panelFilters } = searchState;
```